Sources/Player/PlaybackEngine/Internal/Events/PlayerEventSender.swift

```diff
@@ -209,11 +209,17 @@ private extension PlayerEventSender {
 				print("StreamingPrivilegesHandler failed to get credentials")
 				return
 			}
-
+			
+			let userClientId: Int? = if let userClientIdSupplier {
+				userClientIdSupplier()
+			} else {
+				nil
+			}
```

Comment:
This could be cleaner like this

```suggestion
			let userClientId = userClientIdSupplier?()
```